File: backend/src/main/java/com/PC2/backend/entity/Curso.java
package com.PC2.backend.entity;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "cursos")
public class Curso {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, length = 120)
	private String nombre;

	@Column(nullable = false, length = 500)
	private String descripcion;

	@Column(nullable = false)
	private Integer cupos;

	@OneToMany(mappedBy = "curso", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
	private List<Matriculado> matriculados = new ArrayList<>();
}

File: frontend/src/app/models/curso.ts
export interface Matriculado {
  id: number;
  nombre: string;
  email: string;
}

export interface Curso {
  id: number;
  nombre: string;
  descripcion: string;
  cupos: number;
  cuposDisponibles: number;
  matriculados: Matriculado[];
}

export interface MatriculadoRequest {
  nombre: string;
  email: string;
}
